src/wallet/wallet.cpp

```diff
@@ -2591,15 +2596,42 @@ void CWallet::AvailableCoins(std::vector<COutput>& vCoins, bool fOnlySafe, const
                 }
                 if(!found) continue;
 
+                if (pcoin->tx->vout[i].nValue < nMinimumAmount || pcoin->tx->vout[i].nValue > nMaximumAmount)
+                    continue;
+
+                if (coinControl && coinControl->HasSelected() && !coinControl->fAllowOtherInputs && !coinControl->IsSelected(COutPoint((*it).first, i)))
+                    continue;
+
+                if (IsLockedCoin((*it).first, i) || nCoinType == ONLY_1000)
```

Comment:
Should be `IsLockedCoin((*it).first, i) && nCoinType != ONLY_1000`

Reply:
resolved